plugin/trino-google-sheets/src/main/java/io/trino/plugin/google/sheets/SheetsClient.java

```diff
@@ -245,6 +275,7 @@ private List<List<Object>> readAllValuesFromSheetExpression(String sheetExpressi
             return sheetsService.spreadsheets().values().get(sheetId, defaultRange).execute().getValues();
         }
         catch (IOException e) {
+            // TODO: improve error to a {Table|Sheet}NotFoundException
```

Comment:
See comment from @ebyhr here: https://github.com/trinodb/trino/pull/15829#discussion_r1092600469

I did not want to change the error code returned from `SHEETS_UNKNOWN_TABLE_ERROR` to `NOT_FOUND` since this would be backwards incompatible. 

Updated the comment to make the reasoning more clear.

Reply:
thanks, now i can read the comment at least :) 

however, i don't understand why we would want to throw "not found" within IOException catch block
to me, the current SHEETS_UNKNOWN_TABLE_ERROR looks fine (the IOException is just some error)

if we want to detect that IOException conveys "specific sheet doesn't exist", we could do that (but we wouldnt' say TableNotFound even then, because the requested sheet may not be a table)


if it was up to me to decide, and if no new information is provided to me, i would just delete this TODO comment and worry about that later, and only if we learn later there is something to worry about